src/pybind/mgr/dashboard/frontend/src/app/shared/components/pwd-expiration-notification/pwd-expiration-notification.component.spec.ts

```diff
@@ -94,6 +95,20 @@ describe('PwdExpirationNotificationComponent', () => {
       });
       component.ngOnInit();
       expect(component['alertType']).toBe('danger');
+      expect(component.displayNotification).toBeTruthy();
+    });
+
+    it('should not display if date is far', () => {
+      const dateValue = Date;
+      spyOn(global, 'Date').and.callFake((date) => {
+        if (date) {
+          return new dateValue(date);
+        } else {
+          return new Date('2022-01-01T00:00:00.000Z');
+        }
+      });
```

Comment:
This can be shorten a bit:
```suggestion
      spyOn(global, 'Date').and.callFake((date) => new dateValue(date ? date : '2022-01-01T00:00:00.000Z'));
```

Reply:
Done. Went a step further and extracted the spy logic into a new method.

Reply:
Thanks haven't seen it was used that often. Maybe putting it into a `beforeEach` would be even better.

Reply:
the date value is different in some cases.